js/main.js

```diff
@@ -0,0 +1,105 @@
+  'Сдается 1-к квартира в ЖК «Летний» на Пулковском шоссе. Квартира сдаётся впервые. Площадь 38 кв м. Лоджия с остеклением до пола. Имеются все условия для комфортного проживания: два шкафа-купе и один полочный (много места для хранения вещей). Из техники: стиральная машина, холодильник, электрический чайник, телевизор, варочная панель, духовая печь, микроволновая печь, WiFi-роутер. Фильтр для очистки воды.',
+  'Новый дом,шведский комплекс «Элланд», закрытая территория, 5 минут до метро! В квартире новый ремонт, вся необходимая техника. На длительный срок от собственника.'
+];
+var PHOTOS = [
+  'http://o0.github.io/assets/images/tokyo/hotel1.jpg',
+  'http://o0.github.io/assets/images/tokyo/hotel2.jpg',
+  'http://o0.github.io/assets/images/tokyo/hotel3.jpg'
+];
+
+var mapPins = document.querySelector('.map__pins');
+var pinTemplate = document.querySelector('#pin').content;
+
+function getRandomIntInclusive(min, max) {
+  min = Math.ceil(min);
+  max = Math.floor(max);
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+function getRandomValueFromArray(arr) {
+  var randomId = Math.floor(Math.random() * arr.length);
+  return arr[randomId];
+}
+
+function generateOffer() {
+  var locationX = getRandomIntInclusive(0, mapPins.clientWidth - 50);
+  var locationY = getRandomIntInclusive(130, 630);
+
+  return {
+    author: {
+      avatar: 'img/avatars/user0' + Math.floor(Math.random() * NUMBER_ADS + 1) + '.png',
+    },
+    offer: {
+      title: getRandomValueFromArray(TITLES),
+      address: locationX + ', ' + locationY,
+      price: getRandomValueFromArray(PRICES),
+      type: getRandomValueFromArray(TYPES),
+      rooms: getRandomValueFromArray(NUMBER_ROOMS),
+      guests: getRandomValueFromArray(GUESTS),
+      checkin: getRandomValueFromArray(CHECKINS),
+      checkout: getRandomValueFromArray(CHECKOUTS),
+      features: getRandomValueFromArray(FEATURES),
+      description: getRandomValueFromArray(DESCRIPTIONS),
+      photos: getRandomValueFromArray(PHOTOS),
+    },
+    location: {
+      x: locationX,
+      y: locationY
+    }
+  };
+}
+
+function generateOffers() {
+  var offers = [];
+  for (var i = 0; i < NUMBER_ADS; i++) {
+    var offer = generateOffer();
+    offers.push(offer);
+  }
+  return offers;
+}
+
+function renderPin(data) {
+  var clonePin = pinTemplate.cloneNode(true);
+
+  var pinElement = clonePin.querySelector('.map__pin');
+  var avatarElement = clonePin.querySelector('img');
+
+  pinElement.style.top = data.location.y + 'px';
+  pinElement.style.left = data.location.x + 'px';
+  pinElement.style.transform = 'translate(-50%, -100%);';
+
+  avatarElement.src = data.author.avatar;
+  avatarElement.alt = data.offer.title;
+
+  return clonePin;
+}
+
+function renderPins() {
+  var offers = generateOffers();
+  var fragment = document.createDocumentFragment();
+
+  for (var i = 0; i < NUMBER_ADS; i++) {
```

Comment:
тут прохода по всем объявлениям и привязываться нужно только к размеру массива. данные уже сгенерированы, работаем с ними
```suggestion
  for (var i = 0; i < offers.length ; i++) {
```